### PR TITLE
Unpin spark requirements

### LINKED ISSUE #2647: Support pandas 2.0 for featuretools[spark]
When pandas is above 2.0, we get [test failures](https://github.com/alteryx/featuretools/actions/runs/7832364459/job/21370894618#step:12:467) that come from doing ps.to_datetime(lti_df["last_time"])

### LINKED ISSUE #2648: Support Woodwork 0.28.0 for featuretools[spark]
#2611 and #2647 Need to be resolved first. We may also have to upgrade numpy and remove support for python 3.8 if, as a result of supporting ww 0.28, we have to bump the min version of numpy.

### PR DESCRIPTION
**Pull request by tamargrey**
closes [2611](https://github.com/alteryx/featuretools/issues/2611), closes #2648, closes #2647, closes https://github.com/alteryx/featuretools/issues/2611, closes https://github.com/alteryx/featuretools/issues/2651

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: featuretools/entityset/entityset.py [hunk 1]
```diff
@@ -1269,11 +1269,15 @@ def add_last_time_indexes(self, updated_dataframes=None):
...
-                            lti_df["last_time_old"] = ps.to_datetime(
-                                lti_df["last_time_old"],
-                            )
                             # TODO: Figure out a workaround for fillna and replace
+                            if lti_df["last_time_old"].dtype != "datetime64[ns]":
+                                lti_df["last_time_old"] = ps.to_datetime(
+                                    lti_df["last_time_old"],
+                                )
+                            if lti_df["last_time"].dtype != "datetime64[ns]":
+                                lti_df["last_time"] = ps.to_datetime(
```
Comment (tamargrey) [inline featuretools/entityset/entityset.py:1278]: The pyarrow upgrade made it such that calling `ps.to_datetime` on a column that is already datetime resulted in errors. We don't have woodwork initialized on lti_df, so I just did a dtype check here.

File: featuretools/primitives/standard/aggregation/num_unique.py [hunk 2]
```diff
@@ -60,6 +60,6 @@ def finalize(s):
 
             return dd.Aggregation(self.name, chunk=chunk, agg=agg, finalize=finalize)
 
-        if self.use_string_for_pd_calc:
+        if self.use_string_for_pd_calc or agg_type == Library.SPARK:
```
Comment (tamargrey) [inline featuretools/primitives/standard/aggregation/num_unique.py:63]: realized that we had the potential for a bug here with spark dataframes if for some reason a user set `use_string_for_pd_calc` to false but used spark dataframes. We need to use the string for spark dataframes.

File: featuretools/tests/testing_utils/mock_ds.py [hunk 3]
```diff
@@ -36,6 +37,9 @@ def make_ecommerce_entityset(with_integer_time_index=False):
          \\ /   .
           L     Log
     """
+    dask = pytest.importorskip("dask", reason="Dask not installed, skipping")
+    dask.config.set({"dataframe.convert-string": False})
```
Comment (tamargrey) [inline featuretools/tests/testing_utils/mock_ds.py:41]: Setting this dask config allows the `object` dtype to be used by Woodwork, which allows LatLong columns to work and avoids an error with the Equal and NotEqual primitives, which were erroring if a `string[pyarrow]` column was called with a datetime column

File: docs/source/api_reference.rst [hunk 4]
```diff
@@ -28,17 +28,6 @@ Deep Feature Synthesis
...
-Wrappers
-~~~~~~~~
-.. currentmodule:: featuretools
-
-scikit-learn (BETA)
--------------------
-.. autosummary::
-    :toctree: generated/
-
-    wrappers.DFSTransformer
```
Comment (tamargrey) [inline docs/source/api_reference.rst:40]: Removing the DFSTransformer from our docs as it only supports python 3.7 and 3.8 and is causing docs build failures

**Review by thehomebrewnerd — APPROVED** [review]
LGTM!

**Review by thehomebrewnerd — APPROVED** [review]

**Pull request merged**